common/main/cpp/native_c4replicator.cc

```diff
@@ -166,49 +166,49 @@ static bool pullFilterFunction(C4CollectionSpec, C4String, C4String, C4RevisionF
 
 static bool pushFilterFunction(C4CollectionSpec, C4String, C4String, C4RevisionFlags, FLDict, void *);
 
-// ??? Move the field stuff into initialization.
-static bool fromJavaReplColls(
+// ??? Move the field introspection into initialization.
+static int fromJavaReplColls(
         JNIEnv *env,
-        int nColls,
         jobjectArray jColls,
-        C4ReplicationCollection colls[],
+        std::vector<C4ReplicationCollection> colls,
         C4ReplicatorMode mode) {
     jclass cls_replColl = env->FindClass("com/couchbase/lite/internal/core/C4ReplicationCollection");
     if (!cls_replColl)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_scope = env->GetFieldID(cls_replColl, "scope", "Ljava/lang/String;");
     if (!f_ReplColl_scope)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_name = env->GetFieldID(cls_replColl, "name", "Ljava/lang/String;");
     if (!f_ReplColl_name)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_push = env->GetFieldID(cls_replColl, "push", "Z");
     if (!f_ReplColl_push)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_pull = env->GetFieldID(cls_replColl, "pull", "Z");
     if (!f_ReplColl_pull)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_options = env->GetFieldID(cls_replColl, "options", "[B");
     if (!f_ReplColl_options)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_hasPushFilter = env->GetFieldID(cls_replColl, "hasPushFilter", "Z");
     if (!f_ReplColl_hasPushFilter)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_hasPullFilter = env->GetFieldID(cls_replColl, "hasPullFilter", "Z");
     if (!f_ReplColl_hasPullFilter)
-        return false;
+        return -1;
 
     jfieldID f_ReplColl_token = env->GetFieldID(cls_replColl, "token", "J");
     if (!f_ReplColl_token)
-        return false;
+        return -1;
 
+    int nColls = env->GetArrayLength(jColls);
```

Comment:
Inside the for loop below, I understand that you would need to add a `C4ReplicationCollection` object to the vector using push_back(), unless the vector is initialized by size, otherwise the behavior could be undefined based on https://cplusplus.com/reference/vector/vector/operator[].

Reply:
Wow.  Thanks!